airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -52,56 +76,277 @@ def get_conn(self):
+        return (
+            self.get_conn()
+            .transferJobs()
+            .get(jobName=job_name, projectId=project_id)
+            .execute(num_retries=NUM_RETRIES)
+        )
+
+    def list_transfer_job(self, filter):
+        """
+        Lists long-running operations in Google Storage Transfer
+        Service that match the specified filter.
+
+        :param filter: (Required) A request filter, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/list#body.QUERY_PARAMETERS.filter
+        :type filter: dict
+        :return: List of Transfer Jobs
+        :rtype: list[dict]
+        """
+        conn = self.get_conn()
+        filter = self._inject_project_id(filter, 'filter project_id')
+        request = conn.transferJobs().list(filter=json.dumps(filter))
+        jobs = []
+
+        while request is not None:
+            response = request.execute(num_retries=NUM_RETRIES)
+            jobs.extend(response['transferJobs'])
+
+            request = conn.transferJobs().list_next(previous_request=request, previous_response=response)
+
+        return jobs
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def update_transfer_job(self, job_name, body):
+        """
+        Updates a transfer job that runs periodically.
+
+        :param job_name: (Required) Name of the job to be updated
+        :type job_name: str
+        :param body: A request body, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/patch#request-body
+        :type body: dict
+        :return: If successful, TransferJob.
+        :rtype: dict
+        """
+        body = self._inject_project_id(body, 'body project_id')
+        return (
+            self.get_conn().transferJobs().patch(jobName=job_name, body=body).execute(num_retries=NUM_RETRIES)
+        )
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    @GoogleCloudBaseHook.catch_http_exception
+    def delete_transfer_job(self, job_name, project_id):
+        """
+        Deletes a transfer job. This is a soft delete. After a transfer job is
+        deleted, the job and all the transfer executions are subject to garbage
+        collection. Transfer jobs become eligible for garbage collection
+        30 days after soft delete.
+
+        :param job_name: (Required) Name of the job to be deleted
+        :type job_name: str
+        :param project_id: (Optional) the ID of the project that owns the Transfer
+            Job. If set to None or missing, the default project_id from the GCP
+            connection is used.
+        :type project_id: str
+        :rtype: None
+        """
+
+        return (
+            self.get_conn()
+            .transferJobs()
+            .patch(
+                jobName=job_name,
+                body={
+                    'project_id': project_id,
+                    'transfer_job': {'status': GcpTransferJobsStatus.DELETED},
```

Comment:
https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs#status

Our goal was to create an operator that delete the job. It was the only known way to do this operation. From the user's perspective, it does not matter if this operation will be reversible or not. The job will be deleted, but it will "be in the trash" can for 30 days.